harpoon/suite.go

```diff
@@ -226,6 +229,28 @@ func (b *SuiteBuilder) Build() (*Suite, error) {
 	opts.DefaultContext = ctx
 	opts.Tags = fmt.Sprintf("~@skip:%s", providerName)
 
+	parsingSuite := &godog.TestSuite{
+		Name:    "acceptance",
+		Options: &opts,
+	}
+	features, err := parsingSuite.RetrieveFeatures()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, feature := range features {
+		if variantTag := featureVariant(feature.Feature.Tags); variantTag != "" {
+			content := bytes.ReplaceAll(feature.Content, []byte("Feature: "), []byte(fmt.Sprintf("Feature: %s - ", variantTag)))
+			content = bytes.ReplaceAll(content, []byte("Scenario: "), []byte(fmt.Sprintf("Scenario: %s - ", variantTag)))
+			content = bytes.ReplaceAll(content, []byte("Scenario Outline: "), []byte(fmt.Sprintf("Scenario Outline: %s - ", variantTag)))
+			content = bytes.ReplaceAll(content, []byte(fmt.Sprintf("@variant:%s", variantTag)), []byte(fmt.Sprintf("@injectVariant:%s", variantTag)))
+			opts.FeatureContents = append(opts.FeatureContents, godog.Feature{
+				Name:     path.Join(variantTag, feature.Uri),
+				Contents: content,
+			})
+		}
+	}
```

Comment:
This is the main hacky bit. We parse all the features twice, once here and then again when we run things. Basically what this is doing is:

- grabbing all the features
- seeing if any have a `variant` tag
- if it does, duping it and munging the `Scenario/Feature/Scenario Outline` descriptions so the names of the generated tests are different
- replacing the `variant` tag with an internal `injectVariant` tag that tracks the current variant and can be fetched with `t.Variant` at test runtime
- synthetically adding this as an additional feature via `opts.FeatureContents`